client/ayon_core/hosts/maya/plugins/publish/extract_pointcache.py

```diff
@@ -41,66 +81,122 @@ def process(self, instance):
         start = float(instance.data.get("frameStartHandle", 1))
         end = float(instance.data.get("frameEndHandle", 1))
 
-        attrs = instance.data.get("attr", "").split(";")
-        attrs = [value for value in attrs if value.strip()]
+        attribute_values = self.get_attr_values_from_data(
+            instance.data
+        )
+
+        attrs = [
+            attr.strip()
+            for attr in attribute_values.get("attr", "").split(";")
+            if attr.strip()
+        ]
         attrs += instance.data.get("userDefinedAttributes", [])
         attrs += self.bake_attributes
         attrs += ["cbId"]
 
-        attr_prefixes = instance.data.get("attrPrefix", "").split(";")
-        attr_prefixes = [value for value in attr_prefixes if value.strip()]
+        attr_prefixes = [
+            attr.strip()
+            for attr in attribute_values.get("attrPrefix", "").split(";")
+            if attr.strip()
+        ]
         attr_prefixes += self.bake_attribute_prefixes
 
+        user_attrs = [
+            attr.strip()
+            for attr in attribute_values.get("userAttr", "").split(";")
+            if attr.strip()
+        ]
+
+        user_attr_prefixes = [
+            attr.strip()
+            for attr in attribute_values.get("userAttrPrefix", "").split(";")
+            if attr.strip()
+        ]
+
         self.log.debug("Extracting pointcache..")
         dirname = self.staging_dir(instance)
 
         parent_dir = self.staging_dir(instance)
         filename = "{name}.abc".format(**instance.data)
         path = os.path.join(parent_dir, filename)
 
-        options = {
-            "step": instance.data.get("step", 1.0),
-            "attr": attrs,
-            "attrPrefix": attr_prefixes,
-            "writeVisibility": True,
-            "writeCreases": True,
-            "writeColorSets": instance.data.get("writeColorSets", False),
-            "writeFaceSets": instance.data.get("writeFaceSets", False),
-            "uvWrite": True,
-            "selection": True,
-            "worldSpace": instance.data.get("worldSpace", True)
-        }
-
+        root = None
         if not instance.data.get("includeParentHierarchy", True):
             # Set the root nodes if we don't want to include parents
             # The roots are to be considered the ones that are the actual
             # direct members of the set
-            options["root"] = roots
+            root = roots
 
-        if int(cmds.about(version=True)) >= 2017:
-            # Since Maya 2017 alembic supports multiple uv sets - write them.
-            options["writeUVSets"] = True
+        kwargs = {
```

Comment:
Are we lacking the usage of the python and mel per frame and post job callbacks? If so, let's fix and also include in testing notes that's something we want to specifically test?

Reply:
Good spot! Have added some extra missing ones.